airbyte-cdk/bulk/core/load/src/testFixtures/kotlin/io/airbyte/cdk/load/discover/DiscoverIntegrationTest.kt

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2025 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.load.discover
+
+import io.airbyte.cdk.command.ConfigurationSpecification
+import io.airbyte.cdk.command.FeatureFlag
+import io.airbyte.cdk.load.command.Property
+import io.airbyte.cdk.load.test.util.ConfigurationUpdater
+import io.airbyte.cdk.load.test.util.FakeConfigurationUpdater
+import io.airbyte.cdk.load.test.util.FakeDataDumper
+import io.airbyte.cdk.load.test.util.IntegrationTest
+import io.airbyte.cdk.load.test.util.NoopDestinationCleaner
+import io.airbyte.cdk.load.test.util.NoopExpectedRecordMapper
+import io.airbyte.cdk.load.test.util.destination_process.NonDockerizedDestinationFactory
+import io.airbyte.protocol.models.v0.AirbyteConnectionStatus
+import io.airbyte.protocol.models.v0.AirbyteMessage
+import java.util.regex.Pattern
+import kotlin.test.assertEquals
+import kotlinx.coroutines.runBlocking
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertAll
+
+data class DiscoverTestConfig(
+    val configContents: String,
+    val featureFlags: Set<FeatureFlag> = emptySet(),
+    val name: String? = null,
+)
+
+abstract class DiscoverIntegrationTest<T : ConfigurationSpecification>(
+    val successConfigFilenames: List<DiscoverTestConfig>,
+    val failConfigFilenamesAndFailureReasons: Map<DiscoverTestConfig, Pattern>,
+    additionalMicronautEnvs: List<String> = emptyList(),
+    micronautProperties: Map<Property, String> = emptyMap(),
+    configUpdater: ConfigurationUpdater = FakeConfigurationUpdater,
+) :
+    IntegrationTest(
+        additionalMicronautEnvs = additionalMicronautEnvs,
+        dataDumper = FakeDataDumper,
+        destinationCleaner = NoopDestinationCleaner,
+        recordMangler = NoopExpectedRecordMapper,
+        configUpdater = configUpdater,
+        micronautProperties = micronautProperties,
+    ) {
+    @BeforeEach
+    fun setupProcessFactory() {
```

Comment:
nit: you can delete this, it's only needed in CheckIntegrationTest

(we're doing some sketchy stuff in `check`, so have to hack around it. But you're not doing that in discover afaict, so it should be fine to skip this)

Reply:
deleted